.github/workflows/generate-client.yaml

```diff
@@ -0,0 +1,116 @@
+name: Generate Client and Create PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch_name:
+        description: "Branch name for the PR (optional)"
+        required: false
+        default: "generate-client"
+      pr_title:
+        description: "PR title (optional)"
+        required: false
+        default: "Update generated TypeScript client"
+      pr_body:
+        description: "PR description (optional)"
+        required: false
+        default: "Automated update of the generated TypeScript client from OpenAPI spec"
+
+jobs:
+  generate-client:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Not sure how all of the CI stuff is wired up, but I noticed in the [backport-ce ](https://github.com/hashicorp/vault/blob/main/.github/workflows/backport-ce.yml#L25)the token is `secrets.ELEVATED_GITHUB_TOKEN`